test/foundry/BaseStrategy.t.sol

```diff
@@ -0,0 +1,200 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.10;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "forge-std/Test.sol";
+
+import "../../contracts/strategies/BaseStrategy.sol";
+import "../../contracts/mocks/MockStakingReward.sol";
+
+import "../../contracts/interfaces/IERC20Detailed.sol";
+
+contract TestStrategy is BaseStrategy {
+    function initialize(
+        string memory _name,
+        string memory _symbol,
+        address _strategyToken,
+        address _underlyingToken,
+        address _owner
+    ) public initializer {
+        _initialize(_name, _symbol, _strategyToken, _underlyingToken, _owner);
+    }
+
+    /// @dev makes the actual deposit into the `strategy`
+    /// @param _amount amount of tokens to deposit
+    function _deposit(uint256 _amount) internal override returns (uint256) {
+        underlyingToken.approve(strategyToken, _amount);
+        MockStakingReward(strategyToken).stake(_amount);
+        return _amount;
+    }
+
+    /// @dev makes the actual withdraw from the 'strategy'
+    /// @return amountWithdrawn returns the amount withdrawn
+    function _withdraw(uint256 _amountToWithdraw, address _destination)
+        internal
+        override
+        returns (uint256)
+    {
+        MockStakingReward(strategyToken).unstake(_amountToWithdraw);
+        underlyingToken.transfer(_destination, _amountToWithdraw);
+        return _amountToWithdraw;
+    }
+
+    function _redeemRewards(bytes calldata)
+        internal
+        override
+        returns (uint256 underlyingReward, uint256[] memory)
+    {
+        // uint256 _amountToWithdraw = abi.decode(data, (uint256));
+        uint256 balBefore = underlyingToken.balanceOf(address(this));
+        MockStakingReward(strategyToken).claimRewards();
+        underlyingReward = underlyingToken.balanceOf(address(this)) - balBefore;
+    }
+
+    function getRewardTokens() external view returns (address[] memory) {}
+}
+
+contract TestBaseStrategy is Test {
```

Comment:
can we inherit, or somehow reuse, the `TestIdleCDOBase.sol` and instead do something like this https://github.com/Idle-Labs/idle-tranches/blob/master/test/foundry/IdleClearpoolStrategy.t.sol to avoid code repetitions? Feel free to add also the missing `testOnlyIdleCDO` and `testOnlyOwner` in the TestIdleCDOBase